utils/reviewsUtils.js

```diff
@@ -48,28 +46,22 @@ export async function postReview(cmsAdapter, review) {
   await cmsAdapter.postReview(review);
 }
 
-async function getMovieReview(id) {
-  const res = await fetch(`${API_BASE}/reviews?filters[movie]=${id}`);
-  const payload = await res.json();
-  const modifiedArr = payload.data.map((obj) => ({
+export async function getAverageRating(adapter, id) {
+  const payload = await adapter.loadMoviesReviews(id);
+  const modifiedArr = payload.map((obj) => ({
     id: obj.id,
     ...obj.attributes,
   }));
-  return modifiedArr;
-}
-
-export async function getAverageRating(id) {
-  const reviewsList = await getMovieReview(id);
+  let filteredReviews = filterVerified(modifiedArr);
   const imdbRes = await imdbRating(id);
```

Comment:
This can make unnecessary requests to IMDB. Instead of loading the IMDB rating here, it would be better to only load it if the number of filtered reviews is less than 5. If there are more than five verified reviews, IMDB rating will still be loaded here, and then not used anywhere.